h/accounts/views.py

```diff
@@ -30,7 +30,10 @@ def ajax_form(request, result):
 
     if isinstance(result, httpexceptions.HTTPRedirection):
         request.response.headers.extend(result.headers)
-        result = result.json
+        if result.body != '':
```

Comment:
I take it that accessing `result.json` raises an exception? Could it be better to do a try and catch the exception and set result to `{}` in the `except`.